### PR TITLE
Lots of beta stuff

### PR DESCRIPTION
**Pull request by jamesrichford**
# Description
Lots of work for beta closes the following issues  #181, #54, #176 and #83

# Checklist

- [x] I am an awesome developer and proud of my code
- [x] I added / updated / removed relevant unit or integration tests to prove my change works
- [x] I ran all tests using ```npm test``` to make sure everything else still works
- [x] I ran ```npm run review``` to ensure the code adheres to the repository standards

### CODE REVIEW

File: cli/cli-test-runner.ts [hunk 1]
```diff
@@ -8,10 +11,54 @@ export class CliTestRunner {
...
+
+   public run(userArguments: AlsatianCliOptions) {
+
+      // if version has been requested then output the current version and exit
+      if (userArguments.versionRequested) {
+         process.stdout.write("alsatian version " + Package.version);
+         return;
+      }
+
+      // if help has been requested then output the current version and exit
```
Comment (Jameskmonger) [inline cli/cli-test-runner.ts]: Probably worth fixing this comment here @jamesrichford ;)

Comment (jamesrichford) [inline cli/cli-test-runner.ts]: haha that was supposed to read the CLI options, their descriptions and the current version... dear me :P #copyandpastemonster I'll fix that tonight - good spot

File: core/test-set.ts [hunk 2]
```diff
@@ -1,44 +1,51 @@
...
+      const globHelper = new GlobHelper();
+      return new TestSet(testLoader, globHelper);
+   }
 
-    if (typeof testsFileLocations === "string") {
-      testsFileLocations = [ <string>testsFileLocations ];
-    }
+   public addTestsFromFiles (testFileLocation: string): void
+   public addTestsFromFiles (testFileLocations: Array<string>): void
+   public addTestsFromFiles (testsFileLocations: string | Array<string>) {
```
Comment (jamesrichford) [inline core/test-set.ts]: I believe this is to handle older versions of TypeScript that don't support this very well, in that they need to have the intended uses and also a union of all of them. This seems to work nicely for 2.0 but not so much for lower so wanted to be nice to the peeps who haven't got to upgrade yet :)

File: core/test-set.ts [hunk 3]
```diff
@@ -1,44 +1,51 @@
...
 
-  public get testFixtures(): Array<ITestFixture> {
-    return this._testFixtures;
-  }
+   public get testFixtures(): Array<ITestFixture> {
+      return this._testFixtures;
+   }
 
-  public constructor(private _testLoader: TestLoader, private _globHelper: GlobHelper) { }
+   public constructor(private _testLoader: TestLoader, private _globHelper: GlobHelper) { }
```
Comment (jamesrichford) [inline core/test-set.ts]: The private assignment?

Comment (Jameskmonger) [inline core/test-set.ts]: Yeah. I thought you preferred doing this:

    private _testLoader: TestLoader;
    public constructor (testLoader: TestLoader) {
        this._testLoader = testLoader;
    }

Comment (jamesrichford) [inline core/test-set.ts]: Yeah, they prefer it here at new place so just got into the swing of it haha. Just clarifying :) I'll put in a change for that

**Pull request merged**